.github/workflows/ci.yml

```diff
@@ -22,9 +22,17 @@ on:
     branches:
       - main
       - release/**
+    paths:
+      - 'python/**'
+      - '!python/Makefile'
+      - 'crates/**'
```

Comment:
we shouldn't exclude Makefile as some CI jobs may make use of the commands there. code and project setup/dependencies related files should be CI-covered. some example files that can be exempted are like README, NOTICE, LICENSE, etc.

Reply:
Addressed in [174894c](https://github.com/apache/hudi-rs/pull/93/commits/174894c1d22b9de9fc3d4d1786859c9a5c65062a). The CI should now run whenever a change is made in either the rust or python source, setup/project dependencies or changes in test coverage configuration. Although redundant as the paths spec should take care of this, I've explicitly ignored running CI on paths for markdown files and changes in the issue template or pull request template.

Reply:
hmmm this causes maintenance burden as project evolves - we have to keep this list updated so that CI will trigger, note that we already have crates-level Cargo.toml, what about pyproject.toml? wouldn't just defining `paths-ignore` serve the purpose?

Reply:
Fair point. My reasoning was that as the project evolves, majority of the changes would be within the `crates/` and `python/` paths. The trade-off here would be between maintaining a list of paths to trigger the CI on vs. maintaining the list of paths to ignore.
But if non-project dependencies won't change as much, I agree it would be easier to just ignore those instead. I'll make the required changes.

Reply:
Does it make sense to have `cliff.toml` in there as its used to generate changelogs so would be irrelevant for tests?

Reply:
we might need CI to auto generate changelog in future so let's keep it out

Reply:
Please feel free to resolve this and proceed if everything looks okay.